tests/integration/snapshot/v3_snapshot_test.go

```diff
@@ -139,6 +141,9 @@ func TestSnapshotV3RestoreMulti(t *testing.T) {
 
 // TestCorruptedBackupFileCheck tests if we can correctly identify a corrupted backup file.
 func TestCorruptedBackupFileCheck(t *testing.T) {
+	if cpuutil.ByteOrder() == binary.BigEndian {
+		t.Skipf("skipping on big endian platforms since testdata/corrupted_backup.db is in little endian format")
+	}
```

Comment:
This has already been resolved in https://github.com/etcd-io/etcd/pull/20535. Please rebase the PR. 

Please also squash the commits.